src/data/complex/module.lean

```diff
@@ -228,4 +231,61 @@ def conj_ae : ℂ ≃ₐ[ℝ] ℂ :=
 
 @[simp] lemma conj_ae_coe : ⇑conj_ae = conj := rfl
 
+section lift
+
+variables {A : Type*} [ring A] [algebra ℝ A]
+
+/-- There is an alg_hom from `ℂ` to any `ℝ`-algebra with an element that squares to `-1`.
+
+See `complex.lift` for this as an equiv. -/
+def lift_aux (I' : A) (hf : I' * I' = -1) : ℂ →ₐ[ℝ] A :=
+alg_hom.mk'
+  { to_fun := ⇑((algebra.of_id ℝ A).to_linear_map.comp re_lm +
+                (linear_map.to_span_singleton _ _ I').comp im_lm),
+    map_zero' := linear_map.map_zero _,
+    map_add' := linear_map.map_add _,
+    map_mul' := λ ⟨x₁, y₁⟩ ⟨x₂, y₂⟩, by {
+      change algebra_map ℝ A (x₁ * x₂ - y₁ * y₂) + (x₁ * y₂ + y₁ * x₂) • I' =
+             (algebra_map ℝ A x₁ + y₁ • I') * (algebra_map ℝ A x₂ + y₂ • I'),
+      rw [add_mul, mul_add, mul_add, add_comm _ (y₁ • I' * y₂ • I'), add_add_add_comm],
+      congr' 1, -- equate "real" and "imaginary" parts
+      { rw [smul_mul_smul, hf, smul_neg, ←algebra.algebra_map_eq_smul_one, ←sub_eq_add_neg,
+          ←ring_hom.map_mul, ←ring_hom.map_sub], },
+      { rw [algebra.smul_def, algebra.smul_def, algebra.smul_def, ←algebra.right_comm _ x₂,
+          ←mul_assoc, ←add_mul, ←ring_hom.map_mul, ←ring_hom.map_mul, ←ring_hom.map_add], } },
+    map_one' := show algebra_map ℝ A 1 + (0 : ℝ) • I' = 1,
+      by rw [ring_hom.map_one, zero_smul, add_zero], }
+  (linear_map.map_smul _)
```

Comment:
This definition is still verbose, but I can't figure out a nice way to deduplicate it (without bringing in `basis`/`power_basis` like constructions.)

I tried replacing the `to_fun` with the RHS of `lift_aux_apply,` but it seems the elaborator is not smart enough to handle that. In all, I guess this is as good as it gets.

Reply:
Probably this would become a lot nicer if we had an `alg_hom.of_linear_map` constructor.